MonoGame.Framework/Graphics/GraphicsDevice.cs

```diff
@@ -890,6 +893,54 @@ public void SetVertexBuffer(VertexBuffer vertexBuffer, int vertexOffset)
                                    ? _vertexBuffers.Clear()
                                    : _vertexBuffers.Set(vertexBuffer, vertexOffset);
         }
+
+        private void _SetVertexBuffers(VertexBufferBinding[] vertexBuffers, int bindingCount)
+        {
+            if (vertexBuffers == null || vertexBuffers.Length == 0 || bindingCount == 0)
+            {
+                _vertexBuffersDirty |= _vertexBuffers.Clear();
+            }
+            else
+            {
+                if (vertexBuffers.Length > _maxVertexBufferSlots || bindingCount > _maxVertexBufferSlots)

```

Comment:
Same...  only our internal overloads call this so it is never wrong.  Doesn't seem this is needed.